src/pages/instances/actions/ConvertToInstanceBtn.tsx

```diff
@@ -0,0 +1,328 @@
+import { ChangeEvent, FC, useCallback, useEffect, useState } from "react";
+import {
+  ActionButton,
+  Button,
+  Input,
+  Modal,
+  Select,
+} from "@canonical/react-components";
+import { Form, useNavigate } from "react-router-dom";
+import { useFormik } from "formik";
+import * as Yup from "yup";
+import { instanceNameValidation } from "util/instances";
+import { useProject } from "context/project";
+import { createInstance } from "api/instances";
+import usePortal from "react-useportal";
+import { UploadState } from "types/storage";
+import ProgressBar from "components/ProgressBar";
+import {
+  getSupportedArchitectures,
+  humanFileSize,
+  sendFileByWebSocket,
+} from "util/helpers";
+import StoragePoolSelector from "pages/storage/StoragePoolSelector";
+import { useToastNotification } from "context/toastNotificationProvider";
+import { useEventQueue } from "context/eventQueue";
+import { useQueryClient } from "@tanstack/react-query";
+import { queryKeys } from "util/queryKeys";
+import { getInstanceName } from "util/operations";
+
+export interface ConvertToInstanceFormValues {
+  imageFile: File | null;
+  name: string;
+  pool: string;
+  formatConversion: boolean;
+  virtioConversion: boolean;
+  architecture: string;
+}
+
+const CHUNK_SIZE = 1024 * 1024;
+
+const convertToInstancePayload = (values: ConvertToInstanceFormValues) => {
+  const fileSize = values.imageFile?.size || 0;
+  const conversionOptions = [];
+  if (values.formatConversion) {
+    conversionOptions.push("format");
+  }
+  if (values.virtioConversion) {
+    conversionOptions.push("virtio");
+  }
+
+  const instanceArgs = {
+    source: {
+      type: "conversion",
+      mode: "push",
+      conversion_options: conversionOptions,
+    },
+    devices: {
+      root: {
+        path: "/",
+        type: "disk",
+        pool: values.pool,
+      },
+    },
+    type: "virtual-machine",
+    name: values.name,
+    sourceDiskSize: fileSize,
+    architecture: values.architecture,
+  };
+
+  return JSON.stringify(instanceArgs);
+};
+
+const ConvertToInstanceBtn: FC = () => {
+  const { openPortal, closePortal, isOpen, Portal } = usePortal();
+  const { project, isLoading: isProjectLoading } = useProject();
+  const instanceNameAbort = useState<AbortController | null>(null);
+  const [uploadState, setUploadState] = useState<UploadState | null>(null);
+  const [ws, setWs] = useState<WebSocket | null>(null);
+  const toastNotify = useToastNotification();
+  const navigate = useNavigate();
+  const eventQueue = useEventQueue();
+  const queryClient = useQueryClient();
+
+  const handleUploadProgress = (current: number, total: number) => {
+    setUploadState({
+      percentage: Math.floor((current / total) * 100),
+      loaded: current,
+      total: total,
+    });
+
+    if (current === total) {
+      handleCloseModal();
+      navigate(`/ui/project/${project?.name}/instances`);
```

Comment:
I think we should emit an info notification here to inform the upload completed and the import on the server is running in the background.